### PR TITLE
allow lighting kiln (and other lightable furniture) with fire drills [WIP]

### LINKED ISSUE #13211: charcoal kiln (furniture) cannot be activated using camp fire drill
Seems to require matches or lighter.

Presumably the fire drill and camp fire drill ought to work fine though.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25498278-charcoal-kiln-furniture-cannot-be-activated-using-camp-fire-drill?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pipehat**
fixes #13211

### CODE REVIEW

**Comment by Coolthulhu** [comment]
I wouldn't consider this an acceptable fix.

Lighting a cigarette with a fire drill?

**Comment by pipehat** [comment]
Yes, why not? Or we could add a tedious interim step of crafting a tinder nest out of straw or paper and then using that. That seems like pointless simulation to me but I'm not totally against it. If we did that, it's probably a fire starter item with 1 charge that auto-transforms to ash or nothing in a turn or two.

**Comment by Coolthulhu** [comment]
If it is really important, you could either have a separate function for "slow" sources of fire (like fire drill) or just add it to the kiln function.

The `has/use_fire` functions are for sources of fire you can use right now, in negligible time, for things like cauterizing wounds, lighting molotovs and fire arrows, lighting cigarettes and heating up drugs. You can't use a fire drill to cook up meth.

**Comment by pipehat** [comment]
Yeah that's a fair distinction. The extra time-cost of the fire drills ought to be applied here somehow.

**Comment by Coolthulhu** [comment]
The problem with applying it here is that you can't just add a move cost, because it would be in tens of turns. You'd run into the same problem as with opening a gate/soldering, where lighting a molotov with only a fire drill would nail you down to one spot while zombies chew your face off.

Integrating an extended action into `use_fire` is not feasible.

So you could either add a separate long action just to lit a kiln with the fire drill, hardcode the kiln (and only the kiln) to use fire drills in no extra time (in this case the extra move cost is pretty much negligible, because it is out of combat and only few minutes) or use the extra survival tools mod with its ember holder.

**Comment by chaosvolt** [comment]
Yeah, I'd say that emplaced things like kilns should be lightable, but not cigarettes and molotovs.

Any other lightable furniture that would warrant this?

**Comment by pipehat** [comment]
Ugh. Looks like this ideally should plug into the iuse_actor system with maybe a new actor like fire_appliance_igniter and fire_appliance_igniter_extended.

That would let it work with lenses too when weather appropriate. But that's a more involved change.

Marking as WIP for now.

**Comment by tyrael93** [comment]
Since you're at it could you look into fixing lighters taking half a turn to ignite things?

**Comment by pipehat** [comment]
@tyrael93, I assume you're referring to #12467.
I looked through it. There's  definitely overlap but it's  not trivial. I like the sound of Kevin's 'tiny fire' field as the best long term fix for the balance issues you mentioned.

The plan I have in mind should support all extended firestarter items present and future.

File: src/player.cpp [hunk 1]
```diff
@@ -9065,8 +9065,13 @@ bool player::has_fire(const int quantity) const
...
     } else if (has_charges("zweifire_on", quantity)) {
         return true;
+    } else if (has_charges("fire_drill", quantity)) {
+        return true;
+    } else if (has_charges("fire_drill_large", quantity)) {
+        return true;
     }
-    return false;
+
+     return false;
```
Comment (DLaboratory) [inline src/player.cpp]: A more space is inserted.

**Comment by Coolthulhu** [comment]
Probably won't become mergeable.

**Pull request closed without merging**